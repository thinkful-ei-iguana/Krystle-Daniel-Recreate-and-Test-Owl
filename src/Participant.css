.participant {
  display: flex;
  flex-direction: column;
}

.participant-avatar {
  max-width: 2rem;
}

.participant-avatar img {
  width: 2rem;;
}