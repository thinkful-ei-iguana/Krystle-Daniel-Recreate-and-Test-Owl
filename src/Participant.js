import React from "react";


function Participant(props){
    let statusText = "";
    if (props.onStage === true){
        statusText = "on stage";
    } else if (props.onStage === true){
        statusText = "in session";
    } else {
        statusText = "left session";
    }
    
    return (
        <div className="participant">
             <div className="participant-avatar">
                <img src={props.avatar}></img>
             </div>
            <div className="participant-info">
                <div className="participant-name">
                    {props.name}
                </div>
                <div className="participant-status">
                    {statusText}
                </div>
            </div>
        </div>

    );
}

export default Participant;