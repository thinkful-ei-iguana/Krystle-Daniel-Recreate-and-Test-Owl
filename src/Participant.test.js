import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';

describe('Participant component', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Participant 
          name="name" 
          avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
          inSession="true"
          onStage="false"/>, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Participant 
          name="name" 
          avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
          inSession="true"
          onStage="true"/>, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Participant 
          name="name" 
          avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
          inSession="false"
          onStage="false"/>, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    
  it('renders the UI as expected when in session but not on stage', () => {
    const tree = renderer
      .create(<Participant 
        name="name" 
        avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        inSession="true"
        onStage="false"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected when on stage', () => {
    const tree = renderer
      .create(<Participant 
        name="name" 
        avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        inSession="true"
        onStage="true"/>,)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
  it('renders the UI as expected when not present', () => {
    const tree = renderer
      .create(<Participant 
        name="name" 
        avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        inSession="false"
        onStage="false"/>,)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});