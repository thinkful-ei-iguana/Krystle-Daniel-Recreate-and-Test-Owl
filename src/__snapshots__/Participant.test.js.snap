// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant component renders the UI as expected when in session but not on stage 1`] = `
<div
  className="participant"
>
  <div
    className="participant-avatar"
  >
    <img
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="participant-info"
  >
    <div
      className="participant-name"
    >
      name
    </div>
    <div
      className="participant-status"
    >
      left session
    </div>
  </div>
</div>
`;

exports[`Participant component renders the UI as expected when not present 1`] = `
<div
  className="participant"
>
  <div
    className="participant-avatar"
  >
    <img
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="participant-info"
  >
    <div
      className="participant-name"
    >
      name
    </div>
    <div
      className="participant-status"
    >
      left session
    </div>
  </div>
</div>
`;

exports[`Participant component renders the UI as expected when on stage 1`] = `
<div
  className="participant"
>
  <div
    className="participant-avatar"
  >
    <img
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="participant-info"
  >
    <div
      className="participant-name"
    >
      name
    </div>
    <div
      className="participant-status"
    >
      left session
    </div>
  </div>
</div>
`;
